src/develop/blend_gui.c

```diff
@@ -378,6 +378,52 @@ static void _blendif_scale_print_default(float value, float boost_factor, char *
   snprintf(string, n, "%-5.*f", _blendif_print_digits_default(scaled), scaled * 100.0f);
 }
 
+static int _blendif_are_output_channels_used(const dt_develop_blend_params_t *const blend,
+                                             const dt_develop_blend_colorspace_t cst)
+{
+  const unsigned int mask_inclusive = blend->mask_combine & DEVELOP_COMBINE_INCL;
+  const uint32_t mask = cst == DEVELOP_BLEND_CS_LAB ? DEVELOP_BLENDIF_Lab_MASK & DEVELOP_BLENDIF_OUTPUT_MASK
+                                                    : DEVELOP_BLENDIF_RGB_MASK & DEVELOP_BLENDIF_OUTPUT_MASK;
+  const uint32_t active_channels = blend->blendif & mask;
+  const uint32_t inverted_channels = (blend->blendif >> 16) ^ (mask_inclusive ? mask : 0);
+  const uint32_t cancel_channels = inverted_channels & ~blend->blendif & mask;
+  return active_channels || cancel_channels;
+}
+
+static int _blendif_clean_output_channels(dt_iop_module_t *module)
```

Comment:
I know we have plenty of bad coding like this in the current code, but to me here it is better to use a gboolean to make it clear from start that this is a two state value and not some kind of status.

Reply:
Done and updated `_blendif_are_output_channels_used` also.